azure/resource_groups/app/template.json

```diff
@@ -268,142 +293,223 @@
-            },
-            {
-              "name": "DFE_TEACHERS_PAYMENT_SERVICE_DATABASE_NAME",
-              "value": "[parameters('databaseName')]"
-            },
-            {
-              "name": "APPINSIGHTS_INSTRUMENTATIONKEY",
-              "value": "[reference(resourceId('Microsoft.Resources/deployments', variables('applicationInsightsDeploymentName'))).outputs.instrumentationKey.value]"
-            },
-            {
-              "name": "SECRET_KEY_BASE",
-              "value": "[parameters('SECRET_KEY_BASE')]"
-            },
-            {
-              "name": "BASIC_AUTH_USERNAME",
-              "value": "[parameters('BASIC_AUTH_USERNAME')]"
-            },
-            {
-              "name": "BASIC_AUTH_PASSWORD",
-              "value": "[parameters('BASIC_AUTH_PASSWORD')]"
-            },
-            {
-              "name": "DFE_SIGN_IN_ISSUER",
-              "value": "[parameters('DFE_SIGN_IN_ISSUER')]"
-            },
-            {
-              "name": "DFE_SIGN_IN_REDIRECT_BASE_URL",
-              "value": "[parameters('DFE_SIGN_IN_REDIRECT_BASE_URL')]"
-            },
-            {
-              "name": "DFE_SIGN_IN_IDENTIFIER",
-              "value": "[parameters('DFE_SIGN_IN_IDENTIFIER')]"
-            },
-            {
-              "name": "DFE_SIGN_IN_SECRET",
-              "value": "[parameters('DFE_SIGN_IN_SECRET')]"
-            },
-            {
-              "name": "DFE_SIGN_IN_API_CLIENT_ID",
-              "value": "[parameters('DFE_SIGN_IN_API_CLIENT_ID')]"
-            },
-            {
-              "name": "DFE_SIGN_IN_API_SECRET",
-              "value": "[parameters('DFE_SIGN_IN_API_SECRET')]"
-            },
-            {
-              "name": "DFE_SIGN_IN_API_ENDPOINT",
-              "value": "[parameters('DFE_SIGN_IN_API_ENDPOINT')]"
-            },
-            {
-              "name": "NOTIFY_API_KEY",
-              "value": "[parameters('NOTIFY_API_KEY')]"
-            },
-            {
-              "name": "NOTIFY_TEMPLATE_ID",
-              "value": "[parameters('NOTIFY_TEMPLATE_ID')]"
-            },
-            {
-              "name": "ROLLBAR_ACCESS_TOKEN",
-              "value": "[parameters('ROLLBAR_ACCESS_TOKEN')]"
-            },
-            {
-              "name": "ROLLBAR_ENV",
-              "value": "[parameters('ROLLBAR_ENV')]"
-            },
-            {
-              "name": "GOVUK_VERIFY_ENABLED",
-              "value": "[parameters('GOVUK_VERIFY_ENABLED')]"
-            },
-            {
-              "name": "GOVUK_VERIFY_VSP_HOST",
-              "value": "[concat('https://', variables('vspAppServiceName'), '.azurewebsites.net')]"
-            }
-          ]
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[concat(variables('batPlatformDeploymentUrlBase'),'app-service-linux.json')]",
```

Comment:
I don't think this has all of the changes we made to the app service, which is why it was ejected. Can we do this without going back to the BAT template? If the issue was in sharing the `appSettings`, we can do that using our own template deployment system, which we have for e.g. VSP deployment.